chromium_src/chrome/browser/extensions/extension_tab_util.cc

```diff
@@ -159,10 +159,13 @@ std::unique_ptr<api::tabs::Tab> ExtensionTabUtil::CreateTabObject(
   std::unique_ptr<api::tabs::Tab> tab_object(new api::tabs::Tab);
   tab_object->id.reset(new int(GetTabIdForExtensions(contents)));
   tab_object->index = tab_helper->get_index();
+  if (tab_strip) {
```

Comment:
I think `index` needs to always be handled by `tab_helper` or `tab_strip_model`. If they can ever be out-of-sync it will make things really confusing. In other cases I think the tab_helper method returns the value from tab_strip_model